### PR TITLE
Added support for PHP 8.1

### PR DESCRIPTION
**Pull request by Koen1999**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description
This PR adds support for PHP 8.1 and `laminas-form ^3` (which was necessary for that)
This PR is also necessary to add PHP 8.1 support to `laminas-file`.

Only some minor code changes were required because of a deprecation warning for `fopen`.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,12 +19,12 @@
...
     },
     "require": {
-        "php": "^7.3 || ~8.0.0",
+        "php": "^7.3 || ~8.0.0 || ~8.1.0",
         "laminas/laminas-stdlib": "^3.2.1"
     },
     "require-dev": {
         "laminas/laminas-coding-standard": "~1.0.0",
-        "laminas/laminas-json": "^2.6.1",
+        "laminas/laminas-json": "^2.6.1 || ^3.0.0",
```
Comment (Ocramius) [inline composer.json]: IMO just bump to `^3.0.0` and call it a day there

File: src/Adapter/Console.php [hunk 2]
```diff
@@ -164,16 +164,24 @@ public function __destruct()
...
+            } catch (ValueError $fileOpenError) {
+                $stream = false;
+            }
+            $error = ErrorHandler::stop();
         }
-        $error = ErrorHandler::stop();
 
         if ($stream === false) {
-            $previous = $fileOpenError ?: $error;
+            if (isset($error)) {
```
Comment (Ocramius) [inline src/Adapter/Console.php]: `$error` should always be defined upfront - the `ErrorHandler::start()` and `ErrorHandler::stop()` calls are too nested

Comment (Ocramius) [reply src/Adapter/Console.php]: In general `isset($variableThatMayNotExist)` is not valid code under `laminas/`

File: src/Adapter/Console.php [hunk 3]
```diff
@@ -164,16 +164,24 @@ public function __destruct()
...
-            $stream = fopen($resource, 'w');
-        } catch (ValueError $fileOpenError) {
+        if (is_null($resource)) {
             $stream = false;
+        } else {
+            ErrorHandler::start();
+            try {
+                $stream = fopen($resource, 'w');
+            } catch (ValueError $fileOpenError) {
+                $stream = false;
```
Comment (Ocramius) [inline src/Adapter/Console.php]: Perhaps pre-assign `$stream = false`? :thinking: 

We can use a `} finally {` block to restore the error handler

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.json (hunk 1 above)
Comment (Koen1999) [reply composer.json]: Alright, done

File: src/Adapter/Console.php (hunk 2 above)
Comment (Koen1999) [reply src/Adapter/Console.php]: I avoided the complex behavour by passing an optional argument to ErrorHandler::start()

File: src/Adapter/Console.php (hunk 3 above)
Comment (Koen1999) [reply src/Adapter/Console.php]: Nice suggestion

**Comment by Koen1999** [comment]
@Ocramius , I also updated the phpDoc signature of the method I edited because it did not reflect the actual behavior of the code that was already there. Nothing was returned.

**Review by Ocramius — APPROVED** [review]
Thanks @Koen1999!

**Pull request merged**